[WIP] VIZ/EDU: how to analyze the forward model with PCA

dengemann opened the pull request:
Here I have some small example to investigate the effective dimensions in the lead field.
The script with its comments should be self-explanatory.
But, overall, I am a bit surprised about the relatively small number of dimensions.
Any feedback for improving the computation approach and correctness would be highly welcome.

cc @larsoner @agramfort @SherazKhan etc.

Here is the output:

<img width="704" alt="screenshot 2018-11-21 13 10 19" src="https://user-images.githubusercontent.com/1908618/48840504-1b5d3680-ed8f-11e8-8d6b-6863bc8f03cf.png">

agramfort commented on examples/forward/plot_forward_rank.py:
```diff
@@ -0,0 +1,88 @@
+    exp_var_ratio = np.array([
+        _get_var_exp(
+            leadfield[np.random.choice(picks, size=n_chans, replace=False)])
+        for _ in range(n_draws)])
```
I don't get why you need to subsample

agramfort commented:
yes but why not keeping all channels? if you add redundant channels it
won't increase the rank

dengemann commented:
So you’re saying that resulting differences in the svd spectra not simply
reflect higher resolution through added sensors but true intrinsic
resolution?
The 99% cumulative variance explained dimensions are about 50, 25 and 18
for grad, mag, eeg if you take all sensors within each sensor type.
On Wed 21 Nov 2018 at 16:36, Alexandre Gramfort <notifications@github.com>
wrote:

> yes but why not keeping all channels? if you add redundant channels it
> won't increase the rank
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/5736#issuecomment-440708451>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB0fiq5dJLOsMr3nhvE3H2gensVE7iQYks5uxXL4gaJpZM4Ys_1r>
> .
>

agramfort commented:
yes

dengemann commented:
@agramfort I removed subsampling and updated the comments a bit. I'd eventually like to add some more plots for exploring topographic patterns and visiting the effective rank by source label, but this can also be another PR.

<img width="686" alt="screenshot 2018-11-26 00 34 16" src="https://user-images.githubusercontent.com/1908618/48986264-12fc5700-f113-11e8-8192-3f54932677ff.png">

agramfort commented:
maybe @LauriParkkonen or @mshamalainen have an opinion or a ref for the best way to display this.  I am sure this has been done before.

larsoner commented:
@dengemann are you up for polishing this for MRG? If not I'd say let's close and someone can resurrect if they are interested in the issue
